webapp/channels/src/components/analytics/doughnut_chart.tsx

```diff
@@ -16,30 +17,56 @@ type Props = {
 const DoughnutChart: React.FC<Props> = ({title, width, height, data}) => {
     const canvasRef = useRef<HTMLCanvasElement | null>(null);
     const chartRef = useRef<Chart<'doughnut'> | null>(null);
+    const prevData = useRef(data);
 
     useEffect(() => {
-        if (!data || !canvasRef.current) {
-            return;
-        }
+        const isEqual = data === prevData.current || deepEqual(prevData.current, data);
+
+        if (!isEqual) {
+            if (chartRef.current) {
+                chartRef.current.destroy();
+            }
```

Comment:
Maybe I didn't explain myself correctly. We don't want to destroy this on every render. Just when the component unmounts (what you did on line 47).

This could stay similar to what it was with your initial change:
```
if (chartRef.current) {
    chartRef.current.update();
} else {
    const ctx = ...
}
```

Reply:
Ahh sorry.. I was actually looking at this 
```
useEffect(() => {
        return () => {
            chartRef.current?.destroy();
        };
}, []);
```

now, I fixed this!